index.d.ts

```diff
@@ -5,8 +5,22 @@ export type RequestContext = {
   set: <T>(key: string, value: T) => void
 }
 
+export type Hook =
+  | 'preValidation'
+  | 'preHandler'
+  | 'preSerialization'
+  | 'onSend'
+  | 'onResponse'
+  | 'onTimeout'
+  | 'onError'
+  | 'onRoute'
+  | 'onRegister'
+  | 'onReady'
+  | 'onClose'
+
 export type RequestContextOptions = {
   defaultStoreValues?: Record<string, any>
+  hook?: Hook | Hook[]
```

Comment:
I think it is wrong since this doesn't work:

```
fastify.addHook(['onResponse', 'preHandler'], ()=>{}) // throw new FST_ERR_HOOK_INVALID_TYPE()
```

***continue

Reply:
Don't see much point in having multiple hooks, so kept just one.